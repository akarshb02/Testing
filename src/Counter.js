import React, { useState } from 'react'

export default function Counter() {
  const [countData, setCountData] = useState(0);


  const handleClick = () => {
    setCountData(countData + 1);
  }

  const handleDecrement = () => {
    countData > 0 && setCountData(countData - 1)
  }


  return (
    <div data-test="appComponent">
      <h1 className=''>Count number is&nbsp;
        <span data-test='count'>{countData}</span>
      </h1>
      <div>
        <button data-test="button-data" onClick={handleClick} >Increment</button>&nbsp;
        <button data-test="button-decrement" onClick={handleDecrement} >Decrement</button>
      </div>
    </div>
  )
}
