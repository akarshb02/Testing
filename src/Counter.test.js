import Enzyme, { shallow } from 'enzyme'
import EnzymeAdaptor from '@wojtekmaj/enzyme-adapter-react-17';
import Counter from './Counter';



Enzyme.configure({ adapter: new EnzymeAdaptor() })

const setup = () => shallow(<Counter />)

const setupFunction = (wrapper, id) => wrapper.find(`[data-test='${id}']`)

test('component render without error', () => {
  const wrapper = setup();
  const appComponent = setupFunction(wrapper, 'appComponent')
  expect(appComponent.length).toBe(1)

})

test("render increment button", () => {
  const wrapper = setup();
  const appComponent = wrapper.find("[data-test='button-data']")
  expect(appComponent.length).toBe(1)


})

test("counter display", () => {

})

test("counter display starts at 0", () => {

  const wrapper = setup();
  const count = setupFunction(wrapper, "count").text()
  expect(count).toBe("0")

})


test("clicking button it should increment the display", () => {
  const wrapper = setup();
  const buttonClick = setupFunction(wrapper, "button-data")
  buttonClick.simulate("click");
  const displayCount = setupFunction(wrapper, "count").text();
  expect(displayCount).toBe("1")

})


describe("decrement button", () => {
  test("display value should be greater then 0", () => {
    const wrapper = setup();
    const decrementCount = setupFunction(wrapper, "count").text();
    expect(decrementCount).toBe('0');

  })

  test("render decrement button", () => {
    const wrapper = setup();
    const decremenButton = setupFunction(wrapper, "button-decrement");
    expect(decremenButton.length).toBe(1);

  })

  test("button click and display count should decremant by one", () => {
    const wrapper = setup();
    const decrementCount = setupFunction(wrapper, "button-decrement")
    decrementCount.simulate("click")
    const displayCount = setupFunction(wrapper, "count").text();
    expect(displayCount).toBe("0")

  })




})




