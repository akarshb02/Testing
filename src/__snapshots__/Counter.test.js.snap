// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs snapshot 1`] = `
<div
  data-testId="todo-2"
>
  <h1>
    <strike>
      <h1>
        processing
      </h1>
    </strike>
  </h1>
</div>
`;
